backend/src/schema/resolvers/users.spec.js

```diff
@@ -11,51 +11,32 @@ afterEach(async () => {
 })
 
 describe('users', () => {
-  describe('CreateUser', () => {
-    const mutation = `
-      mutation($name: String, $password: String!, $email: String!) {
-        CreateUser(name: $name, password: $password, email: $email) {
-          id
-        }
-      }
-    `
-    describe('given valid password and email', () => {
-      const variables = {
-        name: 'John Doe',
-        password: '123',
-        email: '123@123.de',
-      }
+  describe('User', () => {
+    const query = `query($email: String) { User(email: $email) { id } }`
+    const variables = { email: 'any-email-address@example.org' }
+    beforeEach(() => {
+      client = new GraphQLClient(host)
+    })
 
-      describe('unauthenticated', () => {
-        beforeEach(async () => {
-          client = new GraphQLClient(host)
-        })
+    it('is forbidden', async () => {
+      await expect(client.request(query, variables)).rejects.toThrow('Not Authorised')
+    })
 
-        it('is not allowed to create users', async () => {
-          await expect(client.request(mutation, variables)).rejects.toThrow('Not Authorised')
-        })
+    describe('as admin', () => {
+      beforeEach(async () => {
+        const userParams = {
+          role: 'admin',
+          email: 'admin@example.org',
+          password: '1234',
+        }
+        const factory = Factory()
+        await factory.create('User', userParams)
+        const headers = await login(userParams)
+        client = new GraphQLClient(host, { headers })
       })
 
-      describe('authenticated admin', () => {
-        beforeEach(async () => {
-          const adminParams = {
-            role: 'admin',
-            email: 'admin@example.org',
-            password: '1234',
-          }
-          await factory.create('User', adminParams)
-          const headers = await login(adminParams)
-          client = new GraphQLClient(host, { headers })
-        })
-
-        it('is allowed to create new users', async () => {
-          const expected = {
-            CreateUser: {
-              id: expect.any(String),
-            },
-          }
-          await expect(client.request(mutation, variables)).resolves.toEqual(expected)
-        })
+      it('is permitted', async () => {
+        await expect(client.request(query, variables)).resolves.toEqual({ User: [] })
```

Comment:
would it be better to actually create a user for these test cases, or you are happy that it either throws an authorization error, or not?

Reply:
yes, correct. It's a refactoring but I think I can easily add more meaning to the test.. hang on